### ISSUE TITLE
oneAPI mkl path on windows seems to have changed

### ISSUE THREAD

**Issue by rubu** [kaldi-asr/kaldi#4516]
Seems that the new oneAPI Intel installer places things in different paths than expected by FindBLAS.cmake - after setting everything up (Intel oneAPI 2021.2)  I could not make the cmake process work on Windows and the only way to overcome that was to patch FindBLAS.cmake and add ```lib/${BLAS_mkl_ARCH_NAME}``` to the ```BLAS_mkl_LIB_PATH_SUFFIXES``` variable. Seems that the oneAPI package has changed the structure of things since none of the existing suffixes work, but since I don't have any older MKL installations I can't verify that this breaks anything with the old ones, so just letting you know.

Not sure if this helps much, but this is how the library layout looks like for me after the installation:
![image](https://user-images.githubusercontent.com/1997432/117009939-f35eb500-acf4-11eb-98d3-623280851716.png)

**Comment by kkm000** [comment]
Not only on Windows. oneMKL is a different API, C++ and open source, but includes "classic" closed source MKL in a subdirectory. It's the same MKL, and is being developed and supported (and is the only option for Fortran, so it's not replaced by the new libs), but packaging has changed, and now bundles both libs together.

Unfortunately, our CMake harness barely works, so I cannot provide an ETA for a fix. 

We need to support it in the configure script, tho.

**Comment by jtrmal** [comment]
https://github.com/kaldi-asr/kaldi/pull/4483/files#diff-c956673c6bff360b17cb89a74d59fc5c17b5c2bc915edc5305f56feb788fae0d

**Comment by rubu** [comment]
@kkm000 sure, I don't need an ETA since I did the patch locally and built kaldi, so I'm kinda good. Just wanted to let you know.

**Comment by kkm000** [comment]
@jtrmal, nearly missed your patch, thanks for the x-ref. Will incorporate it, too!

Intel libs install into some system directory. VS integration looks for them in the registry. I never used CMake on Windows, don't even know if it has a built-in registry support. But can do through an external program (reg.exe comes with Windows), if it does not. I think it's the best to just take Intel's own logic for looking up all necessary lib/include paths, in the end.

**Comment by kkm000** [comment]
No, I just hope to switch to CMake eventually. This is certainly helpful
